packages/integrations/tailwind/src/index.ts

```diff
@@ -14,13 +16,59 @@ function getDefaultTailwindConfig(srcUrl: URL) {
 	};
 }
 
-export default function (): AstroIntegration {
+async function getUserConfig(projectRoot: URL, configPath?: string) {
+	const resolvedProjectRoot = fileURLToPath(projectRoot);
+	let userConfigPath: string | undefined;
+
+	if (configPath) {
+		const configPathWithLeadingSlash = /^\.*\//.test(configPath) ? configPath : `./${configPath}`;
+		userConfigPath = fileURLToPath(new URL(configPathWithLeadingSlash, `file://${resolvedProjectRoot}/`));
+	}
+
+	return await load('tailwind', { mustExist: false, cwd: resolvedProjectRoot, filePath: userConfigPath });
+}
+
+type IntegrationConfig =
+	| {
+			config?: {
+				/**
+				 * Path to your tailwind config file
+				 * @default 'tailwind.config.js'
+				 */
+				path?: string;
+				/**
+				 * Apply Astro's default Tailwind config as a preset
+				 * This is recommended to enable Tailwind across all components and Astro files
+				 * @default true
+				 */
+				applyAstroPreset?: boolean;
+			};
+	  }
+	| undefined;
+
+export default function (integrationConfig: IntegrationConfig): AstroIntegration {
+	const applyAstroConfigPreset = integrationConfig?.config?.applyAstroPreset ?? true;
```

Comment:
something I'll keep in mind: setting default variables options seems to be a common use-case for integrations. Nuxt ships with a helper for integrations to define default options easily, maybe could do the same in the future.